bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -430,15 +427,14 @@ STDigest BCStateTran::checkpointReservedPages(uint64_t checkpointNumber, DataSto
   auto numberOfPagesInCheckpoint = pages.size();
   LOG_INFO(getLogger(),
            "Associating pending pages with checkpoint: " << KVLOG(numberOfPagesInCheckpoint, checkpointNumber));
-
+  std::unique_ptr<char[]> buffer(new char[config_.sizeOfReservedPage]);
```

Comment:
Should we zero the newly allocated buffer?

Reply:
There's no need to zero in these cases, since the data are memcpy-ed to them, so I believe it's a bit quicker.
I only zero the buffer in zeroReservedPages().